diff -Naurp linux-d785aef/drivers/amlogic/amports/vvc1.c linux-d785aef-patch/drivers/amlogic/amports/vvc1.c
--- linux-d785aef/drivers/amlogic/amports/vvc1.c	2017-02-23 12:51:45.000000000 +0200
+++ linux-d785aef-patch/drivers/amlogic/amports/vvc1.c	2017-02-23 12:54:49.000000000 +0200
@@ -862,7 +862,7 @@ static void vvc1_local_init(void)
 	/* vvc1_ratio = vvc1_amstream_dec_info.ratio; */
 	vvc1_ratio = 0x100;
 
-	avi_flag = (unsigned long) vvc1_amstream_dec_info.param;
+	avi_flag = (unsigned long) vvc1_amstream_dec_info.param & 0x1;
 	keyframe_pts_only = (u32)vvc1_amstream_dec_info.param & 0x100;
 
 	total_frame = 0;
