diff -Naurp linux-d785aef/drivers/amlogic/hdmi/hdmi_common/hdmi_parameters.c linux-d785aef-patch/drivers/amlogic/hdmi/hdmi_common/hdmi_parameters.c
--- linux-d785aef/drivers/amlogic/hdmi/hdmi_common/hdmi_parameters.c	2016-12-23 11:16:30.000000000 +0200
+++ linux-d785aef-patch/drivers/amlogic/hdmi/hdmi_common/hdmi_parameters.c	2017-02-23 13:01:35.000000000 +0200
@@ -1111,6 +1111,8 @@ struct hdmi_format_para *hdmi_get_fmt_na
 
 	for (i = 0; i < sizeof(all_fmt_paras) /
 		sizeof(struct hdmi_format_para *); i++) {
+		if (!all_fmt_paras[i])
+			break;
 		lname = all_fmt_paras[i]->name;
 		if (lname && (strncmp(name, lname, strlen(lname)) == 0)) {
 			vic = all_fmt_paras[i]->vic;
