diff -Naurp linux-d785aef/arch/arm64/boot/dts/amlogic/mesongxbb-gpu-mali450.dtsi linux-d785aef-patch/arch/arm64/boot/dts/amlogic/mesongxbb-gpu-mali450.dtsi
--- linux-d785aef/arch/arm64/boot/dts/amlogic/mesongxbb-gpu-mali450.dtsi	2017-02-25 17:55:13.000000000 +0200
+++ linux-d785aef-patch/arch/arm64/boot/dts/amlogic/mesongxbb-gpu-mali450.dtsi	2017-02-26 10:53:13.000000000 +0200
@@ -28,7 +28,7 @@
 		num_of_pp = <3> ;
 		def_clk = <3>;
 		sc_mpp = <3>;/* number of pp used most of time.*/
-		tbl = <&clk125_cfg &clk285_cfg &clk400_cfg &clk500_cfg &clk666_cfg &clk800_cfg>;
+		tbl = <&clk125_cfg &clk285_cfg &clk400_cfg &clk500_cfg &clk666_cfg &clk800_cfg &clk800_cfg>;
 
 		clocks = <&clock CLK_FPLL_DIV3>,
 			<&clock CLK_FPLL_DIV4>,
diff -Naurp linux-d785aef/arch/arm64/boot/dts/amlogic/mesongxl-gpu-mali450.dtsi linux-d785aef-patch/arch/arm64/boot/dts/amlogic/mesongxl-gpu-mali450.dtsi
--- linux-d785aef/arch/arm64/boot/dts/amlogic/mesongxl-gpu-mali450.dtsi	2017-02-25 17:55:13.000000000 +0200
+++ linux-d785aef-patch/arch/arm64/boot/dts/amlogic/mesongxl-gpu-mali450.dtsi	2017-02-26 10:53:13.000000000 +0200
@@ -28,7 +28,7 @@
 		num_of_pp = <3> ;
 		def_clk = <3> ;
 		sc_mpp = <3>;/* number of pp used most of time.*/
-		tbl = <&clk125_cfg &clk285_cfg &clk400_cfg &clk500_cfg &clk666_cfg &clk800_cfg>;
+		tbl = <&clk125_cfg &clk285_cfg &clk400_cfg &clk500_cfg &clk666_cfg &clk800_cfg &clk800_cfg>;
 
 		clocks = <&clock CLK_FPLL_DIV3>,
 			<&clock CLK_FPLL_DIV4>,
