diff -Naurp linux-d785aef/sound/soc/aml/m8/aml_spdif_codec.c linux-d785aef-patch/sound/soc/aml/m8/aml_spdif_codec.c
--- linux-d785aef/sound/soc/aml/m8/aml_spdif_codec.c	2016-12-23 11:16:30.000000000 +0200
+++ linux-d785aef-patch/sound/soc/aml/m8/aml_spdif_codec.c	2017-03-04 09:42:39.159217221 +0200
@@ -26,9 +26,8 @@
 
 #define DRV_NAME "spdif-dit"
 
-#define STUB_RATES	SNDRV_PCM_RATE_8000_192000
-#define STUB_FORMATS	(SNDRV_PCM_FMTBIT_S16_LE | \
-	SNDRV_PCM_FMTBIT_S24_LE | SNDRV_PCM_FMTBIT_S32_LE)
+#define STUB_RATES	SNDRV_PCM_RATE_32000 | SNDRV_PCM_RATE_44100 | SNDRV_PCM_RATE_48000 | SNDRV_PCM_RATE_88200 | SNDRV_PCM_RATE_96000 | SNDRV_PCM_RATE_176400 | SNDRV_PCM_RATE_192000
+#define STUB_FORMATS	SNDRV_PCM_FMTBIT_S16_LE
 
 struct pinctrl *pin_spdif_ctl;
 struct device *spdif_dev;
diff -Naurp linux-d785aef/sound/soc/codecs/aml_codec_t9015.c linux-d785aef-patch/sound/soc/codecs/aml_codec_t9015.c
--- linux-d785aef/sound/soc/codecs/aml_codec_t9015.c	2016-12-23 11:16:30.000000000 +0200
+++ linux-d785aef-patch/sound/soc/codecs/aml_codec_t9015.c	2017-03-04 09:42:39.159217221 +0200
@@ -428,10 +428,8 @@ static int aml_T9015_audio_resume(struct
 	return 0;
 }
 
-#define T9015_AUDIO_STEREO_RATES SNDRV_PCM_RATE_8000_96000
-#define T9015_AUDIO_FORMATS (SNDRV_PCM_FMTBIT_S16_LE \
-			| SNDRV_PCM_FMTBIT_S20_3LE | SNDRV_PCM_FMTBIT_S24_LE \
-			| SNDRV_PCM_FMTBIT_S8 | SNDRV_PCM_FMTBIT_S32_LE)
+#define T9015_AUDIO_STEREO_RATES (SNDRV_PCM_RATE_44100 | SNDRV_PCM_RATE_48000 | SNDRV_PCM_RATE_88200 | SNDRV_PCM_RATE_96000 | SNDRV_PCM_RATE_192000)
+#define T9015_AUDIO_FORMATS (SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S32_LE)
 
 struct snd_soc_dai_ops T9015_audio_aif_dai_ops = {
 	.hw_params = aml_T9015_hw_params,
diff -Naurp linux-d785aef/sound/soc/codecs/dummy_codec.c linux-d785aef-patch/sound/soc/codecs/dummy_codec.c
--- linux-d785aef/sound/soc/codecs/dummy_codec.c	2016-12-23 11:16:30.000000000 +0200
+++ linux-d785aef-patch/sound/soc/codecs/dummy_codec.c	2017-03-04 09:42:39.159217221 +0200
@@ -16,9 +16,8 @@ struct dummy_codec_private {
 	struct snd_soc_codec codec;
 };
 
-#define DUMMY_CODEC_RATES		(SNDRV_PCM_RATE_8000_192000)
-#define DUMMY_CODEC_FORMATS		(SNDRV_PCM_FMTBIT_S16_LE |\
-		SNDRV_PCM_FMTBIT_S24_LE | SNDRV_PCM_FMTBIT_S32_LE)
+#define DUMMY_CODEC_RATES		(SNDRV_PCM_RATE_32000 | SNDRV_PCM_RATE_44100 | SNDRV_PCM_RATE_48000 | SNDRV_PCM_RATE_88200 | SNDRV_PCM_RATE_96000 | SNDRV_PCM_RATE_176400 | SNDRV_PCM_RATE_192000)
+#define DUMMY_CODEC_FORMATS		SNDRV_PCM_FMTBIT_S32_LE
 
 static int dummy_codec_pcm_hw_params(struct snd_pcm_substream *substream,
 				     struct snd_pcm_hw_params *params,
diff -Naurp linux-d785aef/sound/soc/codecs/es8323.c linux-d785aef-patch/sound/soc/codecs/es8323.c
--- linux-d785aef/sound/soc/codecs/es8323.c	2017-03-04 09:39:34.000000000 +0200
+++ linux-d785aef-patch/sound/soc/codecs/es8323.c	2017-03-04 09:42:39.159217221 +0200
@@ -619,12 +619,9 @@ static int es8323_set_bias_level(struct
 	return 0;
 }
 
-#define ES8323_RATES (SNDRV_PCM_RATE_8000 | SNDRV_PCM_RATE_11025 | \
-	SNDRV_PCM_RATE_16000 | SNDRV_PCM_RATE_22050 | SNDRV_PCM_RATE_44100 | \
-	SNDRV_PCM_RATE_48000 | SNDRV_PCM_RATE_88200 | SNDRV_PCM_RATE_96000)
+#define ES8323_RATES (SNDRV_PCM_RATE_44100 | SNDRV_PCM_RATE_48000 | SNDRV_PCM_RATE_88200 | SNDRV_PCM_RATE_96000 | SNDRV_PCM_RATE_192000)
 
-#define ES8323_FORMATS (SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S20_3LE | \
-	SNDRV_PCM_FMTBIT_S24_LE)
+#define ES8323_FORMATS (SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S32_LE)
 
 static struct snd_soc_dai_ops es8323_ops = {
 	.hw_params = es8323_pcm_hw_params,
@@ -638,8 +635,8 @@ struct snd_soc_dai_driver es8323_dai = {
 	.name = "es8323-hifi",
 	.playback = {
 	     .stream_name = "Playback",
-	     .channels_min = 1,
-	     .channels_max = 2,
+	     .channels_min = 2,
+	     .channels_max = 8,
 	     .rates = ES8323_RATES,
 	     .formats = ES8323_FORMATS,
 	},
